core/common/test/samples/InstantSamples.kt

```diff
@@ -0,0 +1,360 @@
+        // Finding a period that it would take to get from the starting instant to the ending instant
+        val startInstant = Instant.parse("2024-01-01T02:00:00Z")
+        val endInstant = Instant.parse("2024-03-01T03:15:03Z")
+        // In New York, we find the difference between 2023-12-31 and 2024-02-29, which is just short of two months
+        val periodInNewYork = endInstant.minus(startInstant, TimeZone.of("America/New_York"))
+        check(periodInNewYork == DateTimePeriod(months = 1, days = 29, hours = 1, minutes = 15, seconds = 3))
+        // In Berlin, we find the difference between 2024-01-01 and 2024-03-01, which is exactly two months
+        val periodInBerlin = endInstant.minus(startInstant, TimeZone.of("Europe/Berlin"))
+        check(periodInBerlin == DateTimePeriod(months = 2, days = 0, hours = 1, minutes = 15, seconds = 3))
+    }
+
+    @Test
+    fun plusDateTimeUnit() {
+        // Finding a moment that's later than the starting point by the given length of calendar time
+        val startInstant = Instant.parse("2024-04-05T22:51:45.586Z")
+        val twoYearsLaterInBerlin = startInstant.plus(2, DateTimeUnit.YEAR, TimeZone.of("Europe/Berlin"))
+        check(twoYearsLaterInBerlin == Instant.parse("2026-04-05T22:51:45.586Z"))
+        val twoYearsLaterInSydney = startInstant.plus(2, DateTimeUnit.YEAR, TimeZone.of("Australia/Sydney"))
+        check(twoYearsLaterInSydney == Instant.parse("2026-04-05T23:51:45.586Z"))
+    }
+
+    @Test
+    fun minusDateTimeUnit() {
+        // Finding a moment that's earlier than the starting point by the given length of calendar time
+        val startInstant = Instant.parse("2024-03-28T02:04:56.256Z")
+        val twoYearsEarlierInBerlin = startInstant.minus(2, DateTimeUnit.YEAR, TimeZone.of("Europe/Berlin"))
+        check(twoYearsEarlierInBerlin == Instant.parse("2022-03-28T01:04:56.256Z"))
+        val twoYearsEarlierInCairo = startInstant.minus(2, DateTimeUnit.YEAR, TimeZone.of("Africa/Cairo"))
+        check(twoYearsEarlierInCairo == Instant.parse("2022-03-28T02:04:56.256Z"))
+    }
+
+    @Test
+    fun plusTimeBasedUnit() {
+        // Finding a moment that's later than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
+        val fiveHoursLater = instant.plus(5, DateTimeUnit.HOUR)
+        check(fiveHoursLater.toEpochMilliseconds() == 12 * 60 * 60 * 1000L)
+    }
+
+    @Test
+    fun minusTimeBasedUnit() {
+        // Finding a moment that's earlier than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
+        val fiveHoursEarlier = instant.minus(5, DateTimeUnit.HOUR)
+        check(fiveHoursEarlier.toEpochMilliseconds() == 2 * 60 * 60 * 1000L)
+    }
+
+    @Test
+    @Ignore // only the JVM has the range wide enough
+    fun plusDateTimeUnitLong() {
```

Comment:
I don't think we need different samples for overloads with longs, unless the exhibit a different behavior

Reply:
Likewise, I don't understand the benefit of merging the samples together.

Reply:
Fixed.